daemon/logger/copier_test.go

```diff
@@ -205,6 +224,43 @@ func TestCopierSlow(t *testing.T) {
 	}
 }
 
+func TestCopierWithSized(t *testing.T) {
+	var jsonBuf bytes.Buffer
+	sizedLogger := &TestSizedLoggerJSON{Encoder: json.NewEncoder(&jsonBuf)}
+
+	var stdout bytes.Buffer
+	for i := 0; i < 30; i++ {
+		if _, err := stdout.WriteString(strings.Repeat(".", sizedLogger.BufSize()) + "\n"); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	c := NewCopier(map[string]io.Reader{"stdout": &stdout}, sizedLogger)
+	c.Run()
+	wait := make(chan struct{})
+	go func() {
+		c.Wait()
+		close(wait)
+	}()
+	c.Close()
```

Comment:
I'm not sure this is safe.

Reply:
Maybe for `stdout` we can use an `io.Pipe` and copy into it from a goroutine. Once the goroutine is done we know the copier is done.

Reply:
Good catch! This could instead better conform to its sibling tests:

```go
c := NewCopier(...)
c.Run()
defer c.Close()
wait := make(chan struct{})
go func() {
	c.Wait()
	close(wait)
}()
select {
case <-time.After(150 * time.Millisecond):
	t.Fatal("Copier failed to do its work in 1 second")
case <-wait:
}

```

If this isn't sensible I'm happy to clean this up using an `io.Pipe`.

Reply:
I think a pipe would be better here otherwise we're just assuming that after some small period of time everything is done, which can lead to a flakey test.

Reply:
Gotcha, I'll update this as soon as I have cycles to spare.